### PR TITLE
Custom regular and minimum gas price

### PR DESCRIPTION
**Pull request by alepc253**
### Feature

- [ ] Updated CHANGELOG.md
- [ ] Is this a user submitted feature request?
  - [x] Link to issue: [#464](https://github.com/MyEtherWallet/MyEtherWallet/issues/464)
- [ ] Add test cases or procedure
- [x] Add PR label

### CODE REVIEW

**Comment by alepc253** [comment]
Not that much... If the network doesn't set a minimum gasprice then it uses
gasPrice / 2.
It shouldn't change anything to other networks but RSK

**Comment by alepc253** [comment]
RSK won't work with gasPrice / 2

**Comment by alepc253** [comment]
@kvhnuke could you rethink this? Let me know if you need more info about RSK `minimumGasPrice`

**Comment by alepc253** [comment]
@kvhnuke bad news! this is really necessary to avoid unmined tx because of using "safe low" gas price that will be for sure below `minimumGasPrice`

File: src/store/actions.js [hunk 1]
```diff
@@ -95,6 +95,10 @@ const setGasPrice = function({ commit }, gasPrice) {
   commit('SET_GAS_PRICE', gasPrice);
 };
 
+const setMinimumGasPrice = function({ commit }, minimumGasPrice) {
+  commit('SET_MINIMUM_GAS_PRICE', minimumGasPrice);
+};
+
```
Comment (SteveMieskoski) [inline src/store/actions.js]: If the minimum gas price is located on the network object then there is not reason to also add a new property to the store/state.

File: src/layouts/InterfaceLayout/components/InterfaceNetworkModal/InterfaceNetworkModal.vue [hunk 2]
```diff
@@ -394,6 +394,15 @@ export default {
...
           this.selectedeNtworkName = network.name;
+          this.selectedNetwork = network;
+          this.$store.dispatch(
+            'setGasPrice',
+            Number(network.type.regularGasPrice || 41)
+          );
+          this.$store.dispatch(
+            'setMinimumGasPrice',
+            Number(network.type.minimumGasPrice)
+          );
```
Comment (SteveMieskoski) [inline src/layouts/InterfaceLayout/components/InterfaceNetworkModal/InterfaceNetworkModal.vue]: Let's try not to add a new property (setMinimumGasPrice) to the store/state. (see comment in actions.js)

File: src/store/mutations.js [hunk 3]
```diff
@@ -71,6 +71,11 @@ const SET_GAS_PRICE = function(state, val) {
   store.set('gasPrice', val);
 };
 
+const SET_MINIMUM_GAS_PRICE = function(state, val) {
+  state.minimumGasPrice = val;
+  store.set('minimumGasPrice', val);
+};
+
```
Comment (SteveMieskoski) [inline src/store/mutations.js]: Let's try not to add a new property to the store/state. (see comment in actions.js)

File: src/store/state.js [hunk 4]
```diff
@@ -9,9 +9,14 @@ const network =
...
+const regularGasPrice = network.type.regularGasPrice;
 const gasPrice =
-  store.get('gasPrice') !== undefined ? store.get('gasPrice') : 10;
-
+  store.get('gasPrice') !== undefined
+    ? store.get('gasPrice')
+    : regularGasPrice !== undefined
+    ? regularGasPrice
+    : 41;
+const minimumGasPrice = network.type.minimumGasPrice || 0;
```
Comment (SteveMieskoski) [inline src/store/state.js]: Let's try not to add a new property to the store/state. (see comment in actions.js)

**Comment by alepc253** [comment]
@SteveMieskoski let me know if it's ok now, thanks!

**Comment by alepc253** [comment]
@SteveMieskoski anything else that should be addressed?

**Comment by kvhnuke** [comment]
I think this is something RSK clients should handle, as throw an error if the gasPrice is below certain amount. We are trying to keep things universal and if we implement this, we have to implement it to all the other chains as well, since I personally dont like having custom options on certain networks but not others. If you can figure out the minimum gas prices for all the networks we support I can take another look at it.

**Comment by kvhnuke** [comment]
on the second thought, what if we change the current way of calculating the low, medium, high to
low = gasPrice
medium = gasPrice*1.5
high = gasPrice*2

what are your thoughts on that?

**Comment by alepc253** [comment]
@kvhnuke thanks for your comments. About your first comment, we can't know that for every other chain but we can use the same magic number that is used if gasPrice query fails.
About your second approach, it's not the ideal scenario but is much better than the actual one. If you feel confortable with this solution, we can do this now and maybe iterate on it later if a better option comes up.

**Comment by alepc253** [comment]
@kvhnuke I went for the second approach, let me know if this is ok now

**Comment by kvhnuke** [comment]
Thanks @alepc253

File: src/layouts/InterfaceLayout/components/InterfaceNetworkModal/InterfaceNetworkModal.vue [hunk 5]
```diff
@@ -393,6 +393,10 @@ export default {
       this.$store.dispatch('switchNetwork', network).then(() => {
         this.$store.dispatch('setWeb3Instance').then(() => {
           this.selectedeNtworkName = network.name;
+          this.$store.dispatch(
+            'setGasPrice',
+            Number(network.type.gasPrice || 41)
```
Comment (kvhnuke) [inline src/layouts/InterfaceLayout/components/InterfaceNetworkModal/InterfaceNetworkModal.vue]: I dont think this is necessary!

**Pull request merged**